### PR TITLE
Upgrade for latest version of exo-run

### PR DESCRIPTION
**Pull request by trushton**
@kevgo @hugobho @martinjaime 
Updates the tweets service to use the latest versions of exoservice, and exo-run with Docker. 

Official Docker image has been pushed as originate/exosphere-tweets-service

### CODE REVIEW

File: features/steps/steps.ls [hunk 1]
```diff
@@ -24,12 +24,9 @@ module.exports = ->
...
 
   @Given /^an instance of this service$/, (done) ->
-    port-reservation
-      ..get-port N (@service-port) ~>
-        @exocom.register-service name: 'tweets', port: @service-port
-        @process = new ExoService service-name: 'tweets', exocom-port: @exocom.pull-socket-port, exorelay-port: @service-port
-          ..listen!
-          ..on 'online', -> done!
+    @process = new ExoService exocom-host: 'localhost', service-name: 'tweets', exocom-port: @exocom-port
+      ..listen!
```
Comment (hugobho) [inline features/steps/steps.ls]: We should probably change this to `connect` as well, to match Exorelay. I'll put up a PR for that.

File: features/steps/steps.ls [hunk 2]
```diff
@@ -24,12 +24,9 @@ module.exports = ->
...
+      ..on 'online', -> wait 10, done # Wait for ExoCom to register the service
```
Comment (kevgo) [inline features/steps/steps.ls]: This creates a potentially flaky test. Please add a TODO here to use `MockExoCom.wait-until-service` once that is shipped. I'm currently working on this.

File: service.yml [hunk 3]
```diff
@@ -21,3 +22,8 @@ messages:
     - tweets.not-found
     - tweets.updated
     - tweets.deleted
+
+
+docker:
+  link:
+    - 'mongo'
```
Comment (kevgo) [inline service.yml]: @hugobho do you need this to be `mongo:mongo`?

File: src/server.ls [hunk 4]
```diff
@@ -13,7 +13,8 @@ module.exports =
 
   before-all: (done) ->
     mongo-db-name = "exosphere-mongo-service-#{env}"
-    MongoClient.connect "mongodb://localhost:27017/#{mongo-db-name}", N (mongo-db) ->
+    mongo-address = if env is \test then \localhost else \mongo
```
Comment (kevgo) [inline src/server.ls]: Can we spin up the MongoDB Docker image for tests as well? That's the whole point of using Docker for the SDK. People shouldn't have to install all sorts of databases in order to work on an application.

Comment (trushton) [inline src/server.ls]: Yes we can, I have included that now as part of `bin/spec`, what do you think of it?

**Comment by trushton** [comment]
All comments addressed.

**Comment by trushton** [comment]
All comments addressed.

File: bin/spec [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
 set -e
-
+docker run -dit -p 27017:27017 --name=test-mongo mongo
 if [ "$#" == "0" ]; then
   node_modules/o-tools/bin/lint
   cucumber-js --tags ~@todo
 else
   DEBUG='*,-express:*,-body-parser:*,-finalhandler,-rails-delegate,-text-stream-search' cucumber-js --tags ~@todo "$@"
 fi
+docker rm -f test-mongo
```
Comment (kevgo) [inline bin/spec]: Keep the image running and only start it on line 3 if it isn't already running?

**Pull request closed without merging**